### ISSUE TITLE
Bug report: unicode error with include + bad args with unicode

### ISSUE THREAD

**Issue by Yannig** [ansible/ansible#13964]
##### Ansible Version:

Devel
##### Ansible Configuration:

You can reproduce the issue using the following playbook:

```
ansible-playbook error-parsing/error-parsing-args.yml
```

You can retrieve it at the following location: https://github.com/Yannig/ansible-issue-low-speed/tree/master/error-parsing
##### Environment:

Ubuntu
##### Summary:

Ansible v2 get a unicode error when parsing bad formated module.
##### Steps To Reproduce:

ansible-playbook error-parsing/error-parsing-args.yml
##### Expected Results:

```
fatal: [localhost]: FAILED! => {"failed": true, "reason": "ERROR! this task 'file' has extra params, which is only allowed in the following modules: command, shell, script, include, include_vars, add_host, group_by, set_fact, raw, meta\n\nThe error appears to have been in '/home/yannig/dev/yannig/ansible-issue-low-speed/error-parsing/error-parsing-args-with-include.yml': line 3, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"Something with unicode héhé and a bad args\"\n  ^ here\n"}
```
##### Actual Results:

```
[WARNING]: Error when using <bound method CallbackModule.v2_runner_on_failed of
</home/yannig/dev/yannig/ansible/lib/ansible/plugins/callback/default.CallbackModule object at 0x7f66ef384f90>>: 'ascii' codec can't decode byte 0xc3 in
position 510: ordinal not in range(128)
```

**Comment by Yannig** [comment]
This patch seems to fix this issue:

``` patch
diff --git a/lib/ansible/plugins/callback/__init__.py b/lib/ansible/plugins/callback/__init__.py
index faf04b1..657b91f 100644
--- a/lib/ansible/plugins/callback/__init__.py
+++ b/lib/ansible/plugins/callback/__init__.py
@@ -88,7 +88,7 @@ class CallbackBase:
         if not keep_invocation and self._display.verbosity < 3 and 'invocation' in result:
             del abridged_result['invocation']

-        return json.dumps(abridged_result, indent=indent, ensure_ascii=False, sort_keys=sort_keys)
+        return to_unicode(json.dumps(abridged_result, indent=indent, ensure_ascii=False, sort_keys=sort_keys))

     def _handle_warnings(self, res):
         ''' display warnings, if enabled and any exist in the result '''
```

**Comment by Yannig** [comment]
Another problem is the returned text in this case that is collapsed. If the same error is raised in main.yml, you get the following formated text:

``````
ERROR! 'ignore_error' is not a valid attribute for a Task

The error appears to have been in '/home/yannig/dev/yannig/ansible-issue-low-speed/error-parsing/error-parsing-args.yml': line 6, column 7, but may         
be elsewhere in the file depending on the exact syntax problem.                                                                                             

The offending line appears to be:                                                                                                                           

  tasks:                                                                                                                                                    
    - name: "Something with unicode héhé and a bad args"                                                                                                    
      ^ here                                                                                                                                                ```
``````

**Comment by abadger** [comment]
Fix pushed to devel and stable-2.0 branches.  Should be in the 2.0.1 release when that goes out the door.
